pkg/kube/utils.go

```diff
@@ -195,3 +197,14 @@ func AddLabelsToPodOptionsFromContext(
 		}
 	}
 }
+
+// FindOwnerReferenceMatch returns true if one of the ownerReferences of the object matches the
+// provided UID.
+func FindOwnerReferenceMatch(ownerReferences []metav1.OwnerReference, uid types.UID) bool {
```

Comment:
small comment:
should we call this function `isOwner` or `isUIDOwner` or even `uidInOwnerRefs`. 
Otherwise the change looks good to me.

Reply:
Let me know what you prefer!